### PR TITLE
Bazel CAS stubs

### PR DESCRIPTION
**Pull request by dgdonut**
* Rearranged remexec to bazel directories
* Added CAS stubs for read, write, findmissing functionality
* some tests

### CODE REVIEW

**Comment by dgdonut** [comment]
Ping. This is ready for, and needs, review.

File: bazel/execution/service_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	execResAny := res.GetResponse()
+	if execResAny == nil {
+		t.Errorf("Nil response from operation: %s", res)
+	}
+
+	metadata := remoteexecution.ExecuteOperationMetadata{}
+	execRes := remoteexecution.ExecuteResponse{}
+	err = ptypes.UnmarshalAny(metadataAny, &metadata)
+	if err != nil {
+		t.Errorf("Failed to unmarhal metadata from any: %v", err)
```
Comment (ryancouto) [inline bazel/execution/service_test.go]: *marshal

File: bazel/execution/service_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	}
+	err = ptypes.UnmarshalAny(execResAny, &execRes)
+	if err != nil {
+		t.Errorf("Failed to unmarhal response from any: %v", err)
```
Comment (ryancouto) [inline bazel/execution/service_test.go]: ditto

File: common/proto/proto.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Library for common gRPC protobuf-related tools.
+
```
Comment (ryancouto) [inline common/proto/proto.go]: Think the pkg level comments only work if there isn't a blank line between comment & pkg name

**Review by ryancouto — APPROVED** [review]
lgtm

File: scootapi/server/setup.go [hunk 4]
```diff
@@ -92,8 +93,8 @@ func Defaults() (*ice.MagicBag, jsonconfig.Schema) {
 			return net.Listen("tcp", scootapi.DefaultSched_GRPC)
 		},
 
-		func(l net.Listener) remexec.GRPCServer {
-			return remexec.NewExecutionServer(l)
+		func(l net.Listener) bazel.GRPCServer {
+			return execution.NewExecutionServer(l)
```
Comment (jasonbs10) [inline scootapi/server/setup.go:97]: Does this get used?

File: binaries/scheduler/main.go [hunk 5]
```diff
@@ -63,8 +63,8 @@ func main() {
 			return net.Listen("tcp", *grpcAddr)
 		},
 
-		func(l net.Listener) remexec.GRPCServer {
-			return remexec.NewExecutionServer(l)
+		func(l net.Listener) bazel.GRPCServer {
```
Comment (jasonbs10) [inline binaries/scheduler/main.go]: This may be redundant.

File: binaries/apiserver/main.go [hunk 6]
```diff
@@ -90,8 +93,14 @@ func main() {
 		func(sh *StoreAndHandler) bundlestore.Store {
 			return sh.store
 		},
+		func() (net.Listener, error) {
+			return net.Listen("tcp", *grpcAddr)
+		},
+		func(l net.Listener) bazel.GRPCServer {
```
Comment (jasonbs10) [inline binaries/apiserver/main.go]: This may be redundant.

**Review by jasonbs10 — APPROVED** [review]

**Pull request merged**